### DISCUSSION TITLE
Integrate laravel storage with google drive

### DISCUSSION THREAD

**Discussion by anilherath** [Laravel-Backpack/community-forum#1150] [Q&A (Help)]
I used yaza\laravel-google-drive-storage to enable google drive integration with my backpack pro installation. 
I followed the steps mentioned in https://github.com/yaza-putu/laravel-google-drive-storage to install and configure the storage. 
After that I changed the setupcreateoperation function to enable google drive to the upload field as follows; 
`  protected function setupCreateOperation()
    {
       .......... 

            [
                'name' => 'report_file',
                'label' => 'Report File',
                'type' => 'upload', 
                'upload' => true,
                'hint' => 'Click the "Browse" button to select a file.',
                'withFiles' =>([
                    'disk' => 'google', // the disk where file will be stored
                    'path' => 'uploads', // the path inside the disk where file will be stored
            ])],
           
      .........     
    }`
    
    Now the problem is it gives following error even I enabled and configured the ssl in my local server. 
    
    _cURL error 60: SSL certificate problem: unable to get local issuer certificate (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://oauth2.googleapis.com/token_ 
    
    Then I disabled the verification in configuration file and now it says; 
    
    _Unable to read file from location: aaaaaa.pdf. File not found_ 
    
   in here aaaaaa.pdf is the uploaded file

**Comment by karandatwani92** [comment]
Hey @anilherath 

It seems to be a file reading issue, and I have a few questions to ask:
-  Is the file uploaded successfully using the Backpack's `upload` field with Uploader?
- Did you try file reading outside the Backpack scope?

**Reply by anilherath** [reply]
Thank you very much @karandatwani92. 
- The file was not uploaded to the google drive. But it was work well with local storage. 
- I didn't try to read the file outside the backpack scope

**Reply by anilherath** [reply]
Why is this question closed?

**Reply by karandatwani92** [reply]
Hey @anilherath 

I thought you could reopen it after checking it outside the backpack scope and then checking within the Backpack field.

**Reply by anilherath** [reply]
@karandatwani92 
Actually I couldn't read it outside the backpack scope, since upload to google drive is not working.

**Reply by karandatwani92** [reply]
Sorry, I can't help there. It's a third-party package. First, It has to work for your outside backpack scope; then only it is supposed to be working with a Backpack field. I hope you understand.

**Reply by pxpm** [reply]
Rephrasing a little bit on @karandatwani92 answer, can you do a `Storage::disk('google')->put(your file)` and see your file uploaded to google ? That needs to work first, when that works ok, it should work with Backpack fields, as that is what we do with your file 👍 

Cheers

**Reply by anilherath** [reply]
@karandatwani92 @pxpm Thank you very much for both of you. I'll check that and update the result here.

**Reply by anilherath** [reply]
> Rephrasing a little bit on @karandatwani92 answer, can you do a `Storage::disk('google')->put(your file)` and see your file uploaded to google ? That needs to work first, when that works ok, it should work with Backpack fields, as that is what we do with your file 👍
> 
> Cheers

Hi @pxpm This is the results in CMD; 
c:\wamp64\www\rinsight>php artisan tinker
> Storage::disk('google')->put('cvsample.pdf');

   ArgumentCountError  Too few arguments to function Illuminate\Filesystem\FilesystemAdapter::put(), 1 passed on line 1 and at least 2 expected.

> Storage::disk('google')->put('cvsample.pdf', file_get_contents('cvsample.pdf'));
= `true`

But the file was not uploaded. 
Probably something wrong with the third party app. Is there a direct method to connect backpack pro with google drive?

**Reply by karandatwani92** [reply]
Hey @anilherath

The Backpack is meant only for the admin panel components. It is not positioned to introduce other filesystems.
If you find the solution, please comment below to share it with others.

Closing this one for now. Feel free to reopen if needed.
